sql/core/src/main/scala/org/apache/spark/sql/internal/BaseSessionStateBuilder.scala

```diff
@@ -224,6 +224,7 @@ abstract class BaseSessionStateBuilder(
         TableCapabilityCheck +:
         CommandCheck +:
         CollationCheck +:
+        SyncViewsCheck +:
```

Comment:
The check rules should not come with side effects. Ideally it should check the whole plan and throw exception if something is wrong. However, it seems that there is side effect in this new rule:
```
session.sessionState.catalog.alterTable(updatedViewMeta)
```

Reply:
Unfortunately we don't have an analyzer extension point to run at the very end of the analysis phase. We can add one later, but for now `CheckAnlysis` is the best place to do it. Mabybe we can rename this rule to indicate that it has side effects?

Reply:
"ViewSyncSchemaToMetaStore" coming